bpf/tests/bpf-prog-test-run/bpf-prog-test-run.go

```diff
@@ -0,0 +1,494 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"debug/elf"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"reflect"
+	"strings"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/byteorder"
+	"github.com/cilium/cilium/pkg/maps/ctmap"
+	"github.com/cilium/cilium/pkg/monitor"
+	"github.com/cilium/cilium/pkg/tuple"
+	"github.com/cilium/cilium/pkg/types"
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/perf"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+type ct4GlobalMap map[ctmap.CtKey4Global]ctmap.CtEntry
+
+func ct4TcpMapFromBPF(ebpfMap *ebpf.Map) (ct4GlobalMap, error) {
+	var key ctmap.CtKey4Global
+	var val ctmap.CtEntry
+
+	ret := make(ct4GlobalMap)
+	iter := ebpfMap.Iterate()
+	for iter.Next(&key, &val) {
+		ret[key] = val
+	}
+
+	if err := iter.Err(); err != nil {
+		return ret, fmt.Errorf("Error iterating map: %v", err)
+	}
+
+	return ret, nil
+}
+
+func mapDeleteAll(m *ebpf.Map) error {
+	for {
+		key, err := m.NextKeyBytes(nil)
+		if err != nil {
+			return err
+		}
+		if key == nil {
+			break
+		}
+
+		err = m.Delete(key)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func testNop(spec *ebpf.Collection) error {
+	prog := spec.Programs["test_nop"]
+	if prog == nil {
+		return errors.New("did not find test_nop program")
+	}
+
+	packetIn := gopacket.NewSerializeBuffer()
+	opts := gopacket.SerializeOptions{}
+	gopacket.SerializeLayers(packetIn, opts,
+		&layers.Ethernet{
+			SrcMAC:       net.HardwareAddr{1, 0, 0, 3, 0, 10},
+			DstMAC:       net.HardwareAddr{1, 0, 0, 3, 0, 20},
+			EthernetType: layers.EthernetTypeIPv4,
+		},
+	)
+
+	bpfRet, bufOut, err := prog.Test(packetIn.Bytes())
+	if err != nil {
+		return fmt.Errorf("test run failed: %v", err)
+	}
+
+	if bpfRet != 0 { // CT_ACT_OK
+		return errors.New("unexpected return value")
+	}
+
+	if !reflect.DeepEqual(bufOut, packetIn.Bytes()) {
+		return errors.New("unexpected data modification")
+	}
+
+	return nil
+}
+
+func testMap(spec *ebpf.Collection) error {
+	prog := spec.Programs["test_map"]
+	if prog == nil {
+		return errors.New("did not find test_nop program")
+	}
+
+	bpfCtMap := spec.Maps["test_cilium_ct_tcp4_65535"]
+	if prog == nil {
```

Comment:
```suggestion
	if bpfCtMap == nil {
```

Reply:
:man_facepalming: 

Thanks for catching this :)

Reply:
It should be fixed now.